tests/integration/assignments/test_viewset.py

```diff
@@ -0,0 +1,270 @@
+
+        self.assertEqual(response.status_code, 403)
+
+    def test_withdraw_self(self):
+        self.assignment.set_candidate(get_user_model().objects.get(username='admin'))
+
+        response = self.client.delete(reverse('assignment-candidature-self', args=[self.assignment.pk]))
+
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(Assignment.objects.get(pk=self.assignment.pk).candidates.filter(username='admin').exists())
+        self.assertTrue(Assignment.objects.get(pk=self.assignment.pk).blocked.filter(username='admin').exists())
+
+    def test_withdraw_self_twice(self):
+        response = self.client.delete(reverse('assignment-candidature-self', args=[self.assignment.pk]))
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_withdraw_self_when_finished(self):
+        self.assignment.set_candidate(get_user_model().objects.get(username='admin'))
+        self.assignment.set_phase(Assignment.PHASE_FINISHED)
+        self.assignment.save()
+
+        response = self.client.delete(reverse('assignment-candidature-self', args=[self.assignment.pk]))
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_withdraw_self_during_voting(self):
+        self.assignment.set_candidate(get_user_model().objects.get(username='admin'))
+        self.assignment.set_phase(Assignment.PHASE_VOTING)
+        self.assignment.save()
+
+        response = self.client.delete(reverse('assignment-candidature-self', args=[self.assignment.pk]))
+
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(Assignment.objects.get(pk=self.assignment.pk).candidates.exists())
+        self.assertTrue(Assignment.objects.get(pk=self.assignment.pk).blocked.filter(username='admin').exists())
+
+    def test_withdraw_self_during_voting_non_admin(self):
+        self.assignment.set_candidate(get_user_model().objects.get(username='admin'))
+        self.assignment.set_phase(Assignment.PHASE_VOTING)
+        self.assignment.save()
+        admin = get_user_model().objects.get(username='admin')
+        group_staff = admin.groups.get(name='Staff')
+        group_delegates = type(group_staff).objects.get(name='Delegates')
+        admin.groups.add(group_delegates)
+        admin.groups.remove(group_staff)
+
+        response = self.client.delete(reverse('assignment-candidature-self', args=[self.assignment.pk]))
+
+        self.assertEqual(response.status_code, 403)
+
+
+class CandidatureOther(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.client.login(username='admin', password='admin')
+        self.assignment = Assignment.objects.create(title='test_assignment_leiD6tiojigh1vei1ait', open_posts=1)
+        self.user = get_user_model().objects.create_user(
+            username='test_user_eeheekai4Phue6cahtho',
+            password='test_password_ThahXazeiV8veipeePh6')
+
+    def test_invalid_data(self):
```

Comment:
Could you either change the name of the method to 'test_data_empty_dict' (or something else) or add a comment.

The same on the next methods, which also test invalid data